airqo_device_monitor/external/thingspeak.py

```diff
@@ -0,0 +1,63 @@
+import json, requests
+from datetime import datetime, timedelta
+
+from airqo_device_monitor.constants import (
+    DEFAULT_THINGSPEAK_DATA_INTERVAL_DAYS,
+    THINGSPEAK_API_MAX_NUM_RESULTS,
+    THINGSPEAK_CHANNELS_LIST_URL,
+    THINGSPEAK_FEEDS_LIST_URL,
+    MATHWORKS_USER_ID,
+)
+
+
+def get_data_for_channel(channel, start_time=None, end_time=None):
+    if not start_time:
+        start_time = datetime.now() - timedelta(days=DEFAULT_THINGSPEAK_DATA_INTERVAL_DAYS)
+    if not end_time:
+        end_time = datetime.now()
+
+    # convert to string before the loop because this never changes
+    start_time_string = datetime.strftime(start_time,'%Y-%m-%dT%H:%M:%SZ')
+
+    api_url = THINGSPEAK_FEEDS_LIST_URL.format(channel)
+    all_data = []
+
+    while start_time <= end_time:
+        print "starting for {} to {}".format(start_time, end_time)
```

Comment:
either unnecessary or be more explicit e.g. starting what? Maybe: "Begin gathering data between {} and {}..." and move it outside the while loop?

Reply:
that was just there for testing, ill remove